src/Datadog.Trace/AppSec/Waf/NativeBindings/Native.cs

```diff
@@ -3,22 +3,82 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
-#if !NETFRAMEWORK
-#endif
 using System;
 using System.Collections.Generic;
+using System.IO;
+using System.Linq;
 using System.Runtime.InteropServices;
 using System.Text;
+using Datadog.Trace.Logging;
 
 namespace Datadog.Trace.AppSec.Waf.NativeBindings
 {
     internal class Native
     {
-#if !NETFRAMEWORK
-        private const string DllName = "Datadog.Trace.ClrProfiler.Native";
+#if NETFRAMEWORK
+        private const string DllName = "Sqreen.dll";
+#else
+        private const string DllName = "Sqreen";
+#endif
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(Native));
+
+        static Native()
+        {
+#if NETFRAMEWORK
+            var libName = DllName;
+            var runtimeIdPart1 = "win";
 #else
-        private const string DllName = "Datadog.Trace.ClrProfiler.Native.dll";
+            var (runtimeIdPart1, libPrefix, libExt) =
+                Environment.OSVersion.Platform switch
+                {
+                    PlatformID.MacOSX => ("osx", "lib", "dylib"),
+                    PlatformID.Unix => ("linux", "lib", "so"),
+                    PlatformID.Win32NT => ("win", string.Empty, "dll"),
+                    PlatformID.Win32S => ("win", string.Empty, "dll"),
+                    PlatformID.Win32Windows => ("win", string.Empty, "dll"),
+                    PlatformID.WinCE => ("win", string.Empty, "dll"),
+                    PlatformID.Xbox => throw new NotSupportedException(),
```

Comment:
It's a bit annoying that expression style switch doesn't allow regroup multiple cases together, so I've move to the old statement based switch which does allow cases to be regrouped.

It also reminded me that throwing isn't really a good idea in a static constructor, as it will mean any member accessed after static constructor failure will also throw, so I've changed that as well.